sparrow-py/pysrc/sparrow_py/session.py

```diff
@@ -2,17 +2,52 @@
 from sparrow_py import Expr
 from typing import Optional
 import pyarrow as pa
+import pandas as pd
+
+class InvalidSchemaException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
 
 
 class Table(Expr):
     """ "A table expression."""
+    
+    data: list[pa.Table] = []
 
-    def __init__(self, name: str, ffi: _ffi.Expr):
+    def __init__(self, 
+        name: str,
+        time_column_name: str,
+        key_column_name: str,
+        schema: pa.Schema,
+        ffi: _ffi.Expr,
+        subsort_column_name: Optional[str] = None,
+        grouping_column_name: Optional[str] = None
+    ):
         """Create a new table expression."""
         self.name = name
+        self.time_column_name = time_column_name
+        self.key_column_name = key_column_name
+        self.schema = schema.remove_metadata()
+        self.subsort_column_name = subsort_column_name
+        self.grouping_column_name = grouping_column_name
         self._ffi = ffi
+        
+    def add(self, df: pd.DataFrame) -> None:
```

Comment:
I think at least some of this is going to need to be implemented on the rust side. Specifically, we'll probably want Rust to expose `add_parquet` and `add_pyarrow` or something like that. Then the Python side can expose a variety of methods taking different types and converting to calls to one or the other of those methods.

Reply:
Discussed. We'll do this in a follow-up.